src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/NameAnalyzer.cs

```diff
@@ -28,28 +29,35 @@ private static string GetClosestString(string name, IEnumerable<string> availabl
         }
 
         public static void ReportProblems(
-            CompilationAnalysisContext context, 
-            IEnumerable<ActivityFunctionDefinition> availableFunctions, 
-            IEnumerable<ActivityFunctionCall> calledFunctions)
+            CompilationAnalysisContext context,
+            SemanticModel semanticModel,
+            IEnumerable<ActivityFunctionDefinition> functionDefinitions,
+            IEnumerable<ActivityFunctionCall> functionInvocations)
         {
-            foreach (var activityInvocation in calledFunctions)
+            foreach (var invocation in functionInvocations)
             {
-                if (!availableFunctions.Select(x => x.FunctionName).Contains(activityInvocation.Name))
+                // If invocation uses constant and there is no matching function name in function definition, trust the customer for correctness in case they are using <FunctionsInDependencies>true</FunctionsInDependencies>
```

Comment:
Do we have any knowledge into whether the customer has set `FunctionsInDependencies` in their csproj? If we do, I think it would be preferable to only filter out constants in that case.

Reply:
We don't currently have that information. I could see if doing this would be easy to add that check, but if I have to parse their xml I think I'll just leave it. I figure that if they are using a constant it's unlikely to be an incorrect name here anyway. This does only affect the NameAnalyzer.